Formula/grafana.rb

```diff
@@ -23,7 +23,9 @@ def install
 
     cd grafana_path do
       system "go", "run", "build.go", "build"
-      system "yarn", "install"
+
+      ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/lib/node_modules/npm/bin/node-gyp-bin"
```

Comment:
I'd prefer to see a fix in the node formula for this.

Reply:
This is not a node issue. It just yarn not being aware of homebrew and not looking into the `libexec` folder when trying to find `node-gyp` relative to the `node` executable. This is one of the fixes I wanted to upstream, but I didn't found time to do so yet.
(After that the global node-gyp installation fallback will fail because of sandbox restrictions.)

I can revert this fix if you want, because it fixes only an optional development dependency and I'm still not sure if you want to proper fix yarn configuration on a per formula basis or do something like what was proposed in https://github.com/Homebrew/brew/pull/2962.

Reply:
Let's try to either tweak the layout of the node formula or get something upstreamed to yarn.

>it fixes only an optional development dependency 

I assume it just reinstalls node-gyp, right?

Reply:
> I assume it just reinstalls node-gyp, right?

The install `node-gyp` globally fallback is failing to because of sandbox restrictions and us not configuring yarn correctly to work with the sandbox. Compiling native addons with yarn does not work at all inside the homebrew sandbox as of now. It just skips the optional build dependency fsevents in our case.

Reply:
If you set
```
ENV["PREFIX"] = ENV["HOME"]
```
and invoke yarn as `libexec/"bin/yarn"` does it work?

Reply:
It should, but it's a hack to fix the fallback.

It shouldn't be an issue to leave the unimportant optional build only dependency broken for now (it was broken since the formula started using yarn anyway).

The best way forward is to fix all issues with yarn and homebrew properly, which involves both upstreaming fixes and doing build-time configuration for yarn in homebrew.